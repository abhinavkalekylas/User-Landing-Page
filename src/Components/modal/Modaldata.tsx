import React from "react";
import AddUserForm from "../adduser/Adduser";
import ModalTemplate from "../modaltemplate/ModalTemplate";

const Modaldata = ({
  addUserModal,
  closeModal,
}: {
  addUserModal: boolean;
  closeModal: () => void;
}) => {
  return (
    <ModalTemplate
      title_name={"Add User"}
      userModal={addUserModal}
      closeModal={closeModal}
    >
      <AddUserForm closeModal={closeModal} />
    </ModalTemplate>
  );
};

export default Modaldata;
