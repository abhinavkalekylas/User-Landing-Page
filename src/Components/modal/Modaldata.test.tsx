import { shallow, mount } from "enzyme";
import ReactDOM from "react-dom";
import toJson from "enzyme-to-json";
import Modaldata from "./Modaldata";
import { Button } from "react-bootstrap";
import { eventWrapper } from "@testing-library/user-event/dist/utils";

let wrapper: any;
beforeEach(() => {
  wrapper = shallow(<Modaldata showModal={true} handleClose={jest.fn()} />);
});

// snapshot testing
it("renders correctly Modaldata component", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

// Todo Close button
it("test close button", () => {
  // const mockCallBack = jest.fn();
  // wrapper.find(".close_btn").simulate("click");
  // expect(mockCallBack.mock.calls.length).toEqual(1);
});

// test for rendering modal
it("renders modal when showModal is true", async () => {
  //   wrapper.update();
  //   expect(wrapper.find(".title").exists()).toEqual(true);
});
