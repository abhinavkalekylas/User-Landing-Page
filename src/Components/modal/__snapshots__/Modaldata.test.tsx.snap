// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Adduser component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[MockFunction]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      className="title text-black"
    >
      Add User
    </ModalTitle>
  </ModalHeader>
  <AddUser
    handleClose={[MockFunction]}
  />
  <ModalFooter>
    <Button
      active={false}
      className="close_btn"
      color="info"
      disabled={false}
      onClick={[MockFunction]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
